engine/execution/state/state.go

```diff
@@ -300,6 +318,18 @@ func (s *state) SaveExecutionResults(
 	// but it's the closest thing to atomicity we could have
 	batch := badgerstorage.NewBatch(s.db)
 
+	defer func() {
+		// Rollback if an error occurs during batch operations
+		if err != nil {
+			chunks := result.AllChunkDataPacks()
```

Comment:
what if the node dies (someone pulled the power cord) before this line is reached?

Will this inconsistency cause issues when the node restarts?

Reply:
I don't think so. Because if the second operation failed, then the statecommitment is not saved to database, which means the block is not executed. And after restart, this block will be re-executed, and the database will be consistent again.